memcached.c

```diff
@@ -3279,7 +3356,7 @@ static void process_command(conn *c, char *command) {
     c->iovused = 0;
     if (add_msghdr(c) != 0) {
         out_string(c, "SERVER_ERROR out of memory preparing response");
-        return;
+        goto out;
```

Comment:
Hrm...

I think a separate function is more prone to error as it requires diligence in being consistent with the call point for every failure case. As things stand, we default to "an error occurred" and clear than if the command succeeds, so we only need to keep that set properly on success paths.

Let me see if the code can be re-organized without  gotos without being too messy.